src/components/TradeAssetSearch/helpers/customAssetSearch.ts

```diff
@@ -0,0 +1,24 @@
+import type { AssetNamespace, ChainId } from '@shapeshiftoss/caip'
+import { ASSET_NAMESPACE, bscChainId, solanaChainId } from '@shapeshiftoss/caip'
+import { PublicKey } from '@solana/web3.js'
+
+export const isValidSolanaAddress = (contractAddress: string) => {
+  try {
+    const publicKey = new PublicKey(contractAddress)
+    return publicKey !== null
```

Comment:
nitpick: that's fine as-is, but seems to imply `publicKey` could be null, which it will never be - `PublicKey` constructor will either continue with pk data or throw (https://github.com/solana-labs/solana-web3.js/blob/8ea27fcefe938eaec464c41bde15ac44c6821469/src/publickey.ts#L56-L76).

What about aligning with types like so:

```typescript
export const isValidSolanaAddress = (contractAddress: string) => {
  try {
    new PublicKey(contractAddress)
    return true
  } catch (error) {
    // If instantiation fails, it's not a valid Solana address
    return false
  }
}
```

Reply:
Right you are! https://github.com/shapeshift/web/pull/8619/commits/54115e2e99eb3204fc0ae5b36f478cd27edb27b3